lib/modules/datasource/metadata.ts

```diff
@@ -13,22 +13,26 @@ const githubPages = regEx('^https://([^.]+).github.com/([^/]+)$');
 const gitPrefix = regEx('^git:/?/?');
 
 export function massageUrl(sourceUrl: string): string {
-  const parsedUrl = URL.parse(sourceUrl);
-  if (!parsedUrl?.hostname) {
+  const hostname = getHostname(sourceUrl);
+  if (!hostname) {
     return '';
   }
-  if (parsedUrl.hostname.includes('gitlab')) {
+  if (hostname.includes('gitlab')) {
```

Comment:
we should probably use our util platform detection code here instead 🤔

Reply:
@viceice I don't think I'm able to use platform detection here because the current code is massaging URLs that are not from GitHub or GitLab. If an URL is neither one of those platforms, Renovate massages it as a GitHub URL.

Reply:
Renovate won't massage URLs that are not valid. The platform detection method will return null for invalid URLs and URLs not from GitLab or Github.

Reply:
The platform detection method doesn't provide a way to differentiate between valid and invalid URLs.

Reply:
i know, but it uses hostrules to detect platform for arbitrary urls, so we can match more urls (for self-hosted github / gitlab) too

https://github.com/renovatebot/renovate/blob/6ecd1fdc68c47a794e768a5d37fb37380e32d711/lib/util/common.ts#L14

Reply:
I see your point, and if I use `detectPlatform` function there is a change in behavior with URLs that are not from GitLab or GitHub (e.g example.com). Renovate will delete sourceUrl. Is it something that Renovate should do?

Reply:
For example, the existing test `Should handle failed parsing of sourceUrls for other` will fail because sourceUrl is deleted:

https://github.com/scarrupt/renovate/blob/f7394b04ba878d43686763872adbf2a5ed196758/lib/modules/datasource/metadata.spec.ts#L142